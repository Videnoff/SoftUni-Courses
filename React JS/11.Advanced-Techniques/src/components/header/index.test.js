import React from 'react';
import renderer from 'react-test-renderer';
import Header from "./index";
import TestingEnvironment from "../../test-utils/router";


describe('Header Component', function () {
    it('should render authenticated routes', function () {
        const tree = renderer.create(
            <TestingEnvironment value={{
                user: {
                    loggedIn: true,
                    id: '123',
                }
            }}>
                <Header/>
            </TestingEnvironment>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render non-authenticated routes', function () {
        const tree = renderer.create(
            <TestingEnvironment value={{
                user: {
                    loggedIn: false,
                }
            }}>
                <Header/>
            </TestingEnvironment>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});