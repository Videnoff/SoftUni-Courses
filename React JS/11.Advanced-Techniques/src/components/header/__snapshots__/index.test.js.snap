// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render authenticated routes 1`] = `null`;

exports[`Header Component should render non-authenticated routes 1`] = `null`;
